server/container_create.go

```diff
@@ -265,15 +265,30 @@ func setupContainerUser(ctx context.Context, specgen *generate.Generator, rootfs
 	specgen.SetProcessGID(gid)
 	specgen.AddProcessAdditionalGid(gid)
 
-	for _, group := range addGroups {
-		specgen.AddProcessAdditionalGid(group)
-	}
+	supplementalGroupsPolicy := types.SupplementalGroupsPolicy_Merge
+	if sc != nil {
+		supplementalGroupsPolicy = sc.SupplementalGroupsPolicy
+	}
```

Comment:
How about using [`GetSupplementalGroupsPolicy()`](
https://github.com/kubernetes/cri-api/blob/3a66d9d8665454a681e90f6c26b6173c75d5ab53/pkg/apis/runtime/v1/api.pb.go#L1141-L1146) that can resolve default value?

```suggestion
	supplementalGroupsPolicy := sc.GetSupplementalGroupsPolicy()
```